spec/lib/link_checker.rb

```diff
@@ -130,4 +130,36 @@
       expect(other_urls).to match_array(["http://www.example.com", "https://www.abstract.com"])
     end
   end
+
+  describe "find_invalid_govuk_paths" do
+    it "should separate out gov urls, and example urls" do
+      test_paths = ["/test/path", "/another/test/path", "/a/third/test/path", "/fifth", "/sixth", "/seventh"]
+      test_urls = test_paths.map { |path| "http://www.gov.uk#{path}" }
+
+      mock_client = instance_double("GdsApi::PublishingApi")
+      expect(GdsApi::PublishingApi).to receive(:new).with("https://publishing-api.publishing.service.gov.uk", bearer_token: "test_token") {
+        mock_client
+      }
+      expect(mock_client).to receive(:lookup_content_ids).with({ base_paths: test_paths, with_drafts: true }).and_return(
+        {
+          "/fifth": nil,
+          "/sixth": nil,
+          "/seventh": nil,
+        },
+      )
```

Comment:
I think this is just testing the output of `lookup_content_ids` and not what we do with them so I don't think these expectations are needed.

If you need to stub the output from `lookup_content_ids`, gds-api-adapters has a `stub_publishing_api_has_lookups ` helper method: https://github.com/alphagov/gds-api-adapters/blob/41e9cbf12bec738489340bd9dc63d62427ee3fe7/lib/gds_api/test_helpers/publishing_api.rb#L554